packages/flutter/lib/src/widgets/splash.dart

```diff
@@ -0,0 +1,621 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+
+import 'framework.dart';
+import 'lookup_boundary.dart';
+import 'notification_listener.dart';
+import 'ticker_provider.dart';
+
+// Examples can assume:
+// late Color color;
+// late BuildContext context;
+
+/// Signature for the callback used by [Splash]es to obtain the
+/// appropriate [Rect].
+typedef RectCallback = Rect Function();
+
+/// A splash of [color] shown in response to a user gesture.
+///
+/// The [confirm] and [cancel] methods alter the splash based on the
+/// user's behavior. For example, a tap event might trigger a splash
+/// that's confirmed when the corresponding [GestureDetector.onTapUp]
+/// event is seen and canceled when the pointer is dragged out of the
+/// reference box.
```

Comment:
I actually prefer the two paragraphs from the `InkFeature`


`
/// Subclasses call [confirm] when an input gesture is recognized. For
/// example a press event might trigger an ink feature that's confirmed
/// when the corresponding up event is seen.
///
/// Subclasses call [cancel] when an input gesture is aborted before it
/// is recognized. For example a press event might trigger an ink feature
/// that's canceled when the pointer is dragged out of the reference
/// box.
`

Reply:
> I think what I, as a reader, expect from this doc is how a splash is different from whatever I just draw using normal widget composition.

That makes a lot of sense!

<br>

I restored the two paragraphs you mentioned made a few other updates to the documentation. Thanks for this advice!